contracts/mainnet/connectors/refinance-notional/helpers.sol

```diff
@@ -0,0 +1,243 @@
+		AaveV2LendingPoolProviderInterface(
+			0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5
+		);
+	/**
+	 * @dev Aave V3 Pool Provider
+	 */
+	AaveV3PoolProviderInterface internal constant getAaveV3Provider =
+		AaveV3PoolProviderInterface(0xA55125A90d75a95EC00130E8E8C197dB5641Eb19); //rinkeby address
+
+	/**
+	 * @dev Aave V3 Pool Data Provider
+	 */
+	AaveV3DataProviderInterface internal constant aaveV3Data =
+		AaveV3DataProviderInterface(0x256bBbeDbA70a1240a1EB64210abB1b063267408); //rinkeby address
+
+	/**
+	 * @dev get Referral Code
+	 */
+	uint16 internal constant getReferralCode = 3228;
+
+	struct NotionalBorrowData {
+		// refinance to Notional from source
+		Protocol source;
+		// length of tokens
+		uint256 length;
+		// debt fee
+		uint256 fee;
+		// borrow tokens
+		address[] tokens;
+		// true, then redeems the borrowed balance from cTokens to underlying token before transferring to account
+		bool[] redeemToUnderlying;
+		// borrow amts im underlying token denomination
+		uint256[] amts;
+		// aave V2 borrow rate modes
+		uint256[] rateModes;
+		// fCashAmount accounting for the borrowAmt with debtfee, calculated through SDK
+		uint256[] fCashAmount;
+		// borrow rate max, 0 means any is acceptable
+		uint256[] maxBorrowRate;
+		// notion defined currency IDs of borrowTokens
+		uint256[] currencyIDs;
+		// borrow markets based on the maturity where user wants to borrow
+		uint256[] marketIndex;
+	}
+
+	// withdraw balance of Aave v2
+	function getWithdrawBalanceV2(
+		AaveV2DataProviderInterface aaveData,
+		address token
+	) internal view returns (uint256 bal) {
+		(bal, , , , , , , , ) = aaveData.getUserReserveData(
+			token,
+			address(this)
+		);
+	}
+
+	// withdraw balance of Aave v3
+	function getWithdrawBalanceV3(
+		AaveV3DataProviderInterface aaveData,
+		address token
+	) internal view returns (uint256 bal) {
+		(bal, , , , , , , , ) = aaveData.getUserReserveData(
+			token,
+			address(this)
+		);
+	}
+
+	function getAssetOrUnderlyingToken(uint16 currencyId, bool underlying)
+		internal
+		view
+		returns (address)
+	{
+		// prettier-ignore
+		(Token memory assetToken, Token memory underlyingToken) = notional.getCurrency(currencyId);
+		return
+			underlying ? underlyingToken.tokenAddress : assetToken.tokenAddress;
+	}
+
+	function toUint88(uint256 value) internal pure returns (uint88) {
+		require(value <= type(uint88).max, "uint88 value overflow");
+		return uint88(value);
+	}
+
+	function toUint32(uint256 value) internal pure returns (uint32) {
+		require(value <= type(uint32).max, "uint32 value overflow");
+		return uint32(value);
+	}
+
+	function toUint16(uint256 value) internal pure returns (uint16) {
+		require(value <= type(uint16).max, "uint16 value overflow");
+		return uint16(value);
+	}
+
+	function toUint8(uint256 value) internal pure returns (uint8) {
+		require(value <= type(uint8).max, "uint8 value overflow");
+		return uint8(value);
+	}
```

Comment:
add these in math.sol in common folder

Reply:
Done, also updated the [helpers.sol](https://github.com/Instadapp/dsa-connectors/pull/230/commits/a715112abc5f9da732f31da5301887f851bbf0ea#diff-4da74c75b5284c601e06c717df178e761cbb820aab11de406ef9d6437c5b52a0L74) of notional connector.